### DISCUSSION TITLE
Caching of various Rust crates/dependencies with different features enabled

### DISCUSSION THREAD

**Discussion by felipesere** [mozilla/sccache#2024] [Q&A]
Currently we run `cachepot` (for legacy reasons) and I am keen to get us back on `sccache` in the near future.

One thing that is intriguing to me is that we currently have a cache-per-repo strategy. We did this to avoid any issues that would arise from 2 different projects mixing their cached artefacts.
I am wondering, is this concern valid? 
I feel like if as a company we had a giant cache, it would be a lot warmer and thus effective. But this only makes sense if compiling the same crate with different features/versions does not lead to issues. For example, different projects use tokio in a couple of versions and with a few different features enabled.

**Comment by sylvestre** [comment]
I feel that it is quite unlikely that you will have a sha-256 hash collision.
